test-cloud/test-cloud-performance/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iJbpmEJBTimerPerfIntegrationTest.java

```diff
@@ -0,0 +1,358 @@
+        return () -> {
+            try {
+                long pid = processServicesClient.startProcess(containerId, processId, parameters);
+                assertThat(pid).isNotNull();
+            } catch (KieServicesHttpException e) {
+                logger.error("There has been an error while starting processes", e);
+                throw e;
+            }
+        };
+    }
+    
+    protected void waitForAllProcessesToComplete(Duration waitForCompletionDuration) {
+        BooleanSupplier completionCondition = () -> queryServicesClient.findProcessInstancesByStatus(ACTIVE_STATUS, 0, 1).isEmpty();
+        TimeUtils.wait(waitForCompletionDuration, Duration.of(1, ChronoUnit.SECONDS), completionCondition);
+    }
+    
+    private void gatherAndAssertStatistics() {
+        int numberOfPages = 1 + (PROCESSES_COUNT / 5000);// including one additional page to check there are no more processes
+
+        List<ProcessInstance> completedProcesses = new ArrayList<>(PROCESSES_COUNT);
+
+        for (int i = 0; i < numberOfPages; i++) {
+            List<ProcessInstance> response = queryServicesClient.findProcessInstancesByStatus(Collections.singletonList(org.jbpm.process.instance.ProcessInstance.STATE_COMPLETED), i, 5000);
+            completedProcesses.addAll(response);
+        }
+
+        logger.info("Completed processes count: {}", completedProcesses.size());
+        
+        List<ProcessInstance> activeProcesses = queryServicesClient.findProcessInstancesByStatus(Collections.singletonList(org.jbpm.process.instance.ProcessInstance.STATE_ACTIVE), 0, 100);
+        logger.info("Active processes count: {}", activeProcesses.size());
+        
+        assertThat(activeProcesses).isEmpty();
+        
+        assertThat(completedProcesses).hasSize(PROCESSES_COUNT);
+       
+        Map<String, Integer> startedHostNameDistribution = new HashMap<>();
+        completedHostNameDistribution = new HashMap<>();
+       
+        for (String pod : pods) {
+            completedHostNameDistribution.put(pod, 0);
+            startedHostNameDistribution.put(pod, 0);
+        }
+        
+        for (int i = 0; i < numberOfPages-1; i++) {
+          for (String pod : pods) {
+           int sizeCompleted = queryServicesClient.findProcessInstancesByVariableAndValue("hostName", pod, COMPLETED_STATUS, i, 5000).size();
+           completedHostNameDistribution.put(pod, completedHostNameDistribution.get(pod)+sizeCompleted);
+           
+           int sizeStarted = queryOldValue(i, pod);
+           startedHostNameDistribution.put(pod, startedHostNameDistribution.get(pod)+sizeStarted);
+          }
+        }
+        
+        logger.info("Processes were completed with this distribution: {}", completedHostNameDistribution);
+        logger.info("Processes were started with this distribution: {}", startedHostNameDistribution);       
+    }
+    
+    private void scaleKieServerTo(int count) {
```

Comment:
I am afraid that this is only called if there is more than one pod to be used in scenario. If only 1 pod is used, then the pod's name won't be added to `pods` collection so statistics gathering won't work?

Reply:
right, should be the same for all cases: I've added this case with one pod also to `pods` collection